### PR TITLE
Also export config to JSON

### PR DESCRIPTION
**Pull request by siddhant1**

### CODE REVIEW

File: src/components/header/export-modal/renderer.tsx [hunk 1]
```diff
@@ -77,10 +86,15 @@ class ExportModal extends React.PureComponent<Props, State> {
     ) {
       return;
     }
-    const content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
+    let content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
     const filename = this.state.downloadVegaJSON ? `visualization.vg.json` : `visualization.vl.json`;
+    if (this.state.includeConfig) {
+      content = mergeDeep(content, { config: this.props.config });
```
Comment (domoritz) [inline src/components/header/export-modal/renderer.tsx]: Please confirm that the spec.config has precedence over the config. Also confirm that the behavior is consistent (e.g. make a spec.config and a config that set the background color and see what happens.

Comment (siddhant1) [reply src/components/header/export-modal/renderer.tsx]: Sure!

Comment (siddhant1) [reply src/components/header/export-modal/renderer.tsx]: Made the change!
Findings
1. Even if the config is written , the vega automatically takes the spec.config as of higher priority.
2. The fix includes checking if the config already exists ? dont merge : merge

Comment (domoritz) [reply src/components/header/export-modal/renderer.tsx]: This is incorrect. We specifically want to merge even if the config exists in the spec already.

Comment (siddhant1) [reply src/components/header/export-modal/renderer.tsx]: Should we merge spec.config and config giving spec.config a priority?

Comment (domoritz) [reply src/components/header/export-modal/renderer.tsx]: Yes (if that's consistent with Vega's behavior).

File: src/components/header/export-modal/renderer.tsx [hunk 2]
```diff
@@ -77,10 +86,24 @@ class ExportModal extends React.PureComponent<Props, State> {
...
+    let config = {};
+
+    // Check if the content has the config(i.e spec.config)
+    if (content.config) {
+      // Merge the spec.config and the custom config
+      config = mergeDeep(this.props.config, content.config);
+    }
+
+    if (this.state.includeConfig) {
+      content = mergeDeep(content, { config });
```
Comment (domoritz) [inline src/components/header/export-modal/renderer.tsx]: mergeDeep already merges into the first object no?

File: src/components/header/export-modal/renderer.tsx [hunk 3]
```diff
@@ -77,10 +86,24 @@ class ExportModal extends React.PureComponent<Props, State> {
...
     }
-    const content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
+    let content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
     const filename = this.state.downloadVegaJSON ? `visualization.vg.json` : `visualization.vl.json`;
 
-    const blob = new Blob([JSON.stringify(content, null, 2)], { type: `application/json` });
+    let config = {};
+
+    // Check if the content has the config(i.e spec.config)
+    if (content.config) {
```
Comment (domoritz) [inline src/components/header/export-modal/renderer.tsx]: We shouldn't need this since we merge below, no? Why merge twice?

**Comment by siddhant1** [comment]
I think we are firstly merging both the configs and then merging config with the spec

**Comment by domoritz** [comment]
I don't see why that is necessary.

**Comment by domoritz** [comment]
I think `spec = deepMerge({config}, spec)` is sufficient, no?

**Comment by siddhant1** [comment]
@domoritz I think we can finalize this implementation

File: src/components/header/export-modal/renderer.tsx [hunk 4]
```diff
@@ -80,7 +89,15 @@ class ExportModal extends React.PureComponent<Props, State> {
     const content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
     const filename = this.state.downloadVegaJSON ? `visualization.vg.json` : `visualization.vl.json`;
 
-    const blob = new Blob([JSON.stringify(content, null, 2)], { type: `application/json` });
+    if (this.state.includeConfig) {
+      const Newcontent = mergeDeep({ config: this.props.config }, content as any);
```
Comment (domoritz) [inline src/components/header/export-modal/renderer.tsx]: Variables shouldn't start with capital letters. 

Why didn't the code I suggested not work?

Comment (siddhant1) [reply src/components/header/export-modal/renderer.tsx]: 1. Sure I will change the name of the variable.
2. The second argument takes a priority , so If is pass custom config as first arguement and content (spec.config) as second arguement , the result is perfect , but the config (key) appears on top of the json.
That is why I have deleted the key and added it again , so is appears at the bottom

Comment (domoritz) [reply src/components/header/export-modal/renderer.tsx]: I see. Can you add a comment to explain that?

Comment (siddhant1) [reply src/components/header/export-modal/renderer.tsx]: Sure!

Comment (domoritz) [reply src/components/header/export-modal/renderer.tsx]: I don't see the comment. Can you only respond to my comments once you finished the task? It makes it easier for me to track what has happened and what still needs to be done. Okay?

Comment (siddhant1) [reply src/components/header/export-modal/renderer.tsx]: Okay! pushed changes. Should I make a branch for you to test?

**Pull request merged**